src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -130,6 +168,24 @@ private static (LazyStringValue Doc, LazyStringValue Name) ExtractDocIdAndName(J
             return (doc, name);
         }
 
+        public static (string DocId, string CounterName) ExtractDocIdAndCounterName(LazyStringValue counterTombstoneId)
+        {
+            var p = counterTombstoneId.Buffer;
+            var size = counterTombstoneId.Size;
+                
+            int sizeOfDocId = 0;
+            for (; sizeOfDocId < size; sizeOfDocId++)
+            {
+                if (p[sizeOfDocId] == SpecialChars.RecordSeparator)
+                    break;
+            }
+
+            var doc = Encoding.UTF8.GetString(p, sizeOfDocId);
```

Comment:
In case of big amount of changes and counters, we may end up materializing a lot strings here. I think that we can actually stay with LazyStringValue through all the process. For the dictionary, for an instance, we use the LazyStringValueComparer without having significant changes in our code base.

Reply:
Good idea. Thanks

Reply:
Done for document id in https://github.com/ravendb/ravendb/pull/6713/commits/19390245f79 . We need the counter name as a string anyway